python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -877,6 +900,46 @@ def submit_run_request(run_request) -> SubmitRunRequestResult:
             f"Skipping {skipped_count} {'run' if skipped_count == 1 else 'runs'} for sensor "
             f"{external_sensor.name} already completed with run keys: {seven.json.dumps(run_keys)}"
         )
+
+    # associate these requested runs with the given evaluations, if any
+    yield from _handle_automation_condition_evaluations(
+        run_request_results=gen_run_request_results,
+        evaluations=evaluations,
+        instance=instance,
+        context=context,
+    )
+    yield
+
+
+def _handle_automation_condition_evaluations(
+    run_request_results: Iterator[SubmitRunRequestResult],
+    evaluations: Sequence[AutomationConditionEvaluation],
+    instance: DagsterInstance,
+    context: SensorLaunchContext,
+):
+    if not evaluations:
+        return
+    evaluation_id = int(context.tick_id)
```

Comment:
Would love some insight on how bad of an idea this is (if at all). In the AssetDaemon, we have a ton of logic that handles keeping a monotonically increasing `evaluation_id` across our N `AutomationConditionSensorDefinition` objects.

That seemed quite challenging to implement in the SensorDaemon for a variety of reasons, but I figured we already had a monotonically increasing id available to us in the tick id, so why not just use that?